src/backend/common/util.cpp

```diff
@@ -91,3 +102,115 @@ std::string int_version_to_string(int version) {
     return std::to_string(version / 1000) + "." +
            std::to_string((int)((version % 1000) / 10.));
 }
+
+#if defined(OS_WIN)
+string getTemporaryDirectory() {
+    DWORD bufSize = 261;  // limit according to GetTempPath documentation
+    string retVal;
+    retVal.resize(bufSize);
+    bufSize = GetTempPathA(bufSize, &retVal[0]);
+    retVal.resize(bufSize);
+    return retVal;
+}
+#else
+string getHomeDirectory() {
+    string home = getEnvVar("XDG_CACHE_HOME");
+    if (!home.empty()) return home;
+
+    home = getEnvVar("HOME");
+    if (!home.empty()) return home;
+
+    home = getpwuid(getuid())->pw_dir;
+}
+#endif
+
+bool directoryExists(const string& path) {
+#if defined(OS_WIN)
+    struct _stat status;
+    return _stat(path.c_str(), &status) == 0 && (status.st_mode & S_IFDIR) != 0;
+#else
+    struct stat status;
+    return stat(path.c_str(), &status) == 0 && (status.st_mode & S_IFDIR) != 0;
+#endif
+}
+
+bool createDirectory(const string& path) {
+#if defined(OS_WIN)
+    return CreateDirectoryA(path.c_str(), NULL) != 0;
+#else
+    return mkdir(path.c_str(), 0777) == 0;
+#endif
+}
+
+bool removeFile(const string& path) {
+#if defined(OS_WIN)
+    return DeleteFileA(path.c_str()) != 0;
+#else
+    return unlink(path.c_str()) == 0;
+#endif
+}
+
+bool renameFile(const string& sourcePath, const string& destPath) {
+    return std::rename(sourcePath.c_str(), destPath.c_str()) == 0;
+}
+
+bool isDirectoryWritable(const string& path) {
+    if (!directoryExists(path) && !createDirectory(path)) return false;
+
+    const string testPath = path + AF_PATH_SEPARATOR + "test";
+    if (!std::ofstream(testPath).is_open()) return false;
+    removeFile(testPath);
+
+    return true;
+}
+
+const string& getCacheDirectory() {
+    static std::once_flag flag;
+    static string cacheDirectory;
+
+    std::call_once(flag, []() {
+        const vector<string> pathList = {
+#if defined(OS_WIN)
+            getTemporaryDirectory() + "\\ArrayFire"
+#else
+            "/var/lib/arrayfire",
```

Comment:
Lets remove this entry. I am not sure how to handle this scenario wrt permissions. I think its safe to assume that the home directory is sufficient.

Reply:
Ok. Shall I remove ``/tmp`` as well?

Reply:
I think tmp is fine. There are some cases where a service will be running on a user without a home directory. I can't think of a case where tmp is not available.

Reply:
Ok.